cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/ModelUtil.qll

```diff
@@ -76,19 +85,16 @@ private IndirectReturnOutNode getIndirectReturnOutNode(CallInstruction call, int
  */
 bindingset[d]
 Node callOutput(CallInstruction call, FunctionOutput output, int d) {
-  exists(DataFlow::Node n | n = callOutput(call, output) and d > 0 |
+  exists(DataFlow::Node n, int indirectionIndex |
+    n = callOutputWithIndirectionIndex(call, output, indirectionIndex) and d > 0
+  |
     // The return value
-    result = getIndirectReturnOutNode(n.asInstruction(), d)
+    result = callOutputWithIndirectionIndex(call, output, indirectionIndex + d)
     or
+    n = callOutputWithIndirectionIndex(call, output, indirectionIndex) and
```

Comment:
Why is this line needed? It just seems to repeat the first half of line 89.

Reply:
Oops. You're right. This was a leftover of my quick-eval debugging.

Reply:
Fixed in 1fce26534f7c92ca349038889f7b8c9fd96c6bf6.